cleanlab/rank.py

```diff
@@ -556,52 +556,54 @@ def get_confidence_weighted_entropy_for_each_label(
     return label_quality_scores
 
 
-def get_knn_distance_ood_scores(
-    features: np.array, nbrs: NearestNeighbors, k: int = None
-) -> np.array:
-    """Returns the KNN distance out-of-distribution (OOD) score for each datapoint.
+def get_outlier_scores(features: np.array, knn: NearestNeighbors = None, k: int = 10) -> np.array:
+    """Returns the KNN distance outlier score for each datapoint.
 
-    This is a function to compute OOD scores where higher scores indicate the datapoint is more likely to be OOD.
+    This is a function to compute outlier scores where higher scores indicate the datapoint is more likely to be an outlier.
 
     Parameters
     ----------
     features : np.array
       Feature matrix of shape (N, M), where N is the number of datapoints and M is the number of features.
       This is the "query set" of features for each datapoint which are used for nearest neighbor search.
 
-    nbrs : sklearn.neighbors.NearestNeighbors
+    knn : sklearn.neighbors.NearestNeighbors
       Instantiated NearestNeighbors class object that's been fitted on a dataset in the same feature space.
       Note that the distance metric and n_neighbors is specified when instantiating this class.
+      If knn=None, then by default knn=sklearn.neighbors.NearestNeighbors(n_neighbors=k, metric="cosine").fit(features)
       See: https://scikit-learn.org/stable/modules/neighbors.html
 
-    k : int, default=None
-      Number of neighbors to use when calculating average distance to neighbors.
-      This value k needs to be less than or equal to max_k which is the n_neighbors used when fitting instantiated NearestNeighbors class object.
-      If k=None, then by default k=min(10, max_k) is used where max_k is extracted from the given nbrs.
+    k : int, default=None Number of neighbors to use when calculating average distance to neighbors. This value k
+    needs to be less than or equal to max_k which is the n_neighbors used when fitting instantiated NearestNeighbors
+    class object. If k=None, then by default k=min(10, max_k) is used where max_k is extracted from the given knn.
```

Comment:
> If k=None, then by default k=min(10, max_k) is used where max_k is extracted from the given knn

I actually don't think this is good behavior. If `k = None and knn !=None`, then can we just set `k = n_neighbors` ?  Any other behavior seems counterintuitive to me.

Reply:
Actually good point, for ``k=None`` to be true now ``k=None`` would need to be explicitly passed into the function. ``k=10`` if there is no k specified,  ``k=max_k`` if ``k>n_neighbors`` and ``k=k`` otherwise. Need to update this